### PR TITLE
[RDY] Use handle get buffer refactor

### PR DESCRIPTION
**Pull request by FriedSock**
suggestion from https://github.com/neovim/neovim/issues/6274

### CODE REVIEW

File: src/nvim/buffer.c [hunk 1]
```diff
@@ -1622,7 +1597,7 @@ buf_T * buflist_new(char_u *ffname, char_u *sfname, linenr_T lnum, int flags)
       }
       top_file_num = 1;
     }
-    buf_hashtab_add(buf);
+    handle_register_buffer(buf);
```
Comment (bfredl) [inline src/nvim/buffer.c]: This is already done above.

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -2143,8 +2118,8 @@ buf_T *buflist_findnr(int nr)
     nr = curwin->w_alt_fnum;
   }
 
-  if (pmap_has(handle_T)(buf_map, (handle_T)nr)) {
-    return pmap_get(handle_T)(buf_map, (handle_T)nr);
+  if (handle_exists_buffer((handle_T)nr)) {
```
Comment (bfredl) [inline src/nvim/buffer.c]: This is redundant, `pmap_get`  will return NULL on nonexisting key per convention.

File: src/nvim/buffer.c (hunk 1 above)
Comment (FriedSock) [inline src/nvim/buffer.c]: 👍

File: src/nvim/buffer.c (hunk 2 above)
Comment (FriedSock) [inline src/nvim/buffer.c]: 👍

**Review by bfredl — APPROVED** [review]
LGTM :+1:

**Comment by FriedSock** [comment]
Thank you @bfredl!

**Comment by bfredl** [comment]
Thank you for the patch :)

**Pull request merged**